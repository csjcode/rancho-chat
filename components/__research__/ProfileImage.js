/**
 * 
 * 
 * 
// Tests
// 1. Snapshot test
it('renders correctly', () => {
  const tree = renderer.create(<ProfileImage />).toJSON()
  expect(tree).toMatchSnapshot()
})

// 2. Test that renders with no errors
it('renders without error', () => {
  shallow(<ProfileImage />)
})

// 3. Test for the onPress functionality
it('calls pickImage when pressed', () => {
  const spy = jest.fn()
  const wrapper = shallow(<ProfileImage onPress={spy} />)
  wrapper.find(TouchableOpacity).simulate('press')
  expect(spy).toBeCalled()
})

// 4. Test that the correct image is rendered
it('renders the correct image', () => {
  const wrapper = shallow(<ProfileImage uri={'some_uri'} />)
  expect(wrapper.find('Image').prop('source')).toEqual({ uri: 'some_uri' })
})

// These tests check for basic functionality of the ProfileImage component. The first test checks that the component renders correctly. The second test checks that the component renders with no errors. The third test checks that the onPress function is called when the component is pressed. The fourth test checks that the correct image is rendered.
 * 
 * 
 */
