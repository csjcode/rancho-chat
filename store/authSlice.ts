import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface AuthState {
  token: string | null;
  userData: object | null;
  didTryAutoLogin: boolean;
}

const initialState: AuthState = {
  token: null,
  userData: null,
  didTryAutoLogin: false,
};

const authSlice = createSlice({
  name: "auth",
  initialState: {
    token: null,
    userData: null,
    didTryAutoLogin: false,
  } as AuthState,
  reducers: {
    authenticate: (
      state,
      action: PayloadAction<{ token: string; userData: object }>
    ) => {
      const { payload } = action;
      state.token = payload.token;
      state.userData = payload.userData;
      state.didTryAutoLogin = true;
    },
    setDidTryAutoLogin: (state, action) => {
      state.didTryAutoLogin = true;
    },
    logout: (state, action) => {
      state.token = null;
      state.userData = null;
      state.didTryAutoLogin = false;
    },
    updateLoggedInUserData: (
      state,
      action: PayloadAction<{ newData: object }>
    ) => {
      state.userData = { ...state.userData, ...action.payload.newData };
    },
  },
});

export const setDidTryAutoLogin = authSlice.actions.setDidTryAutoLogin;
export const authenticate = authSlice.actions.authenticate;
export const updateLoggedInUserData = authSlice.actions.updateLoggedInUserData;
export const logout = authSlice.actions.logout;
export default authSlice.reducer;
