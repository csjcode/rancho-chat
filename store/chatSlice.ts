import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface ChatsState {
  chatsData: { [key: string]: any };
}

const initialState: ChatsState = {
  chatsData: {},
};

const chatSlice = createSlice({
  name: "chats",
  initialState,
  reducers: {
    setChatsData: (state, action: PayloadAction<ChatsState>) => {
      state.chatsData = { ...action.payload.chatsData };
    },
  },
});

export const { setChatsData } = chatSlice.actions;
export default chatSlice.reducer;
