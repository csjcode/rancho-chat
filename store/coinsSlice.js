import { createSlice } from '@reduxjs/toolkit'

const coinsSlice = createSlice({
  name: 'coins',
  initialState: {
    storedCoins: {
      coins: ['solana', 'chainlink', 'stepn', 'audius', 'msol'],
    },
  },
  reducers: {
    setStoredCoins: (state, action) => {
      console.log(`action ${JSON.stringify(action)}`)
      const newCoins = action.payload
      state.storedCoins = newCoins
    },
    removeStoredCoin: (state, action) => {
      // {"type":"coins/removeStoredCoin","payload":{"coins":["solana","chainlink","stepn","audius","msol"],"removeCoin":"audius"}}
      console.log(`removeStoredCoin ${JSON.stringify(action)}`)
      const newCoins = action.payload.coins.filter(
        (item) => item !== action.payload.removeCoin,
      )
      // console.log(`action ${JSON.stringify(action)}`)
      // const newCoins = state.storedCoins.coins.pop(action.payload)
      state.storedCoins.coins = newCoins
    },
    addStoredTokenList: (state, action) => {
      console.log(`action ${JSON.stringify(action.payload)}`)
      const newCoins = Array.from(
        new Set([
          ...action.payload.coins,
          ...action.payload.addTokenListStored,
        ]),
      )
      console.log(`newCoins ${newCoins}`)
      state.storedCoins.coins = newCoins
    },
  },
})
export const setStoredCoins = coinsSlice.actions.setStoredCoins
export const removeStoredCoin = coinsSlice.actions.removeStoredCoin
export const addStoredTokenList = coinsSlice.actions.addStoredTokenList
export default coinsSlice.reducer
