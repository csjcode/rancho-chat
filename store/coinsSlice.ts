
import { createSlice } from '@reduxjs/toolkit';

interface StoredCoinsState {
  storedCoins: {
    coins: string[];
  };
}

interface SetStoredCoinsAction {
  payload: StoredCoinsState['storedCoins'];
}

interface RemoveStoredCoinAction {
  payload: {
    coins: string[];
    removeCoin: string;
  };
}

interface AddStoredTokenListAction {
  payload: {
    coins: string[];
    addTokenListStored: string[];
  };
}

const coinsSlice = createSlice({
  name: 'coins',
  initialState: {
    storedCoins: {
      coins: ['solana', 'chainlink', 'stepn', 'audius', 'msol'],
    },
  },
  reducers: {
    setStoredCoins: (state, action: SetStoredCoinsAction) => {
      console.log(`action ${JSON.stringify(action)}`);
      const { payload } = action;
      state.storedCoins = payload;
    },
    removeStoredCoin: (state, action: RemoveStoredCoinAction) => {
      console.log(`removeStoredCoin ${JSON.stringify(action)}`);
      const { payload } = action;
      const { coins, removeCoin } = payload;
      const newCoins = coins.filter((item) => item !== removeCoin);
      state.storedCoins.coins = newCoins;
    },
    addStoredTokenList: (state, action: AddStoredTokenListAction) => {
      console.log(`action ${JSON.stringify(action.payload)}`);
      const { payload } = action;
      const { coins, addTokenListStored } = payload;
      const newCoins = Array.from(
        new Set([...coins, ...addTokenListStored]),
      );
      console.log(`newCoins ${newCoins}`);
      state.storedCoins.coins = newCoins;
    },
  },
});
export const setStoredCoins = coinsSlice.actions.setStoredCoins
export const removeStoredCoin = coinsSlice.actions.removeStoredCoin
export const addStoredTokenList = coinsSlice.actions.addStoredTokenList
export default coinsSlice.reducer
