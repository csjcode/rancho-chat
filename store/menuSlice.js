import { createSlice } from '@reduxjs/toolkit'

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    storedMenu: {
      map: true,
      tricks: true,
      themeColorLight: false,
    },
  },
  reducers: {
    setStoredMenu: (state, action) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload
      state.storedMenu = newMenu
      // console.log(state.storedMenu)
    },
  },
})
export const setStoredMenu = menuSlice.actions.setStoredMenu
export default menuSlice.reducer
