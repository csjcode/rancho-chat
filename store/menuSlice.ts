import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface StoredMenu {
  map: boolean;
  tricks: boolean;
  themeColorsName: string;
}

interface MenuState {
  storedMenu: StoredMenu;
}

const initialState: MenuState = {
  storedMenu: {
    map: true,
    tricks: true,
    themeColorsName: "dark",
  },
};

const menuSlice = createSlice({
  name: "menu",
  initialState,
  reducers: {
    setStoredMenu: (state, action: PayloadAction<StoredMenu>) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload;
      state.storedMenu = newMenu;
    },
  },
});
export const setStoredMenu = menuSlice.actions.setStoredMenu;
export default menuSlice.reducer;

/*


import { createSlice } from '@reduxjs/toolkit'

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    storedMenu: {
      map: true,
      tricks: true,
      themeColorsName: 'dark',
    },
  },
  reducers: {
    setStoredMenu: (state, action) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload
      state.storedMenu = newMenu
    },
  },
})
export const setStoredMenu = menuSlice.actions.setStoredMenu
export default menuSlice.reducer

*/
