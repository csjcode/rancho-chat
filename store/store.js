import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { persistReducer, persistStore } from 'redux-persist'

import AsyncStorage from '@react-native-async-storage/async-storage'
import authSlice from './authSlice'
import chatSlice from './chatSlice'
import coinsSlice from './coinsSlice'
import menuSlice from './menuSlice'
import messagesSlice from './messagesSlice'
import notesSlice from './notesSlice'
import userSlice from './userSlice'

const persistNotesConfig = {
  key: 'notes',
  storage: AsyncStorage,
}

const persistedNotes = persistReducer(persistNotesConfig, notesSlice)

export const store = configureStore({
  reducer: {
    auth: authSlice,
    users: userSlice,
    chats: chatSlice,
    messages: messagesSlice,
    menu: menuSlice,
    coins: coinsSlice,
    notes: persistedNotes,
  },
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
})
export const persistor = persistStore(store)
