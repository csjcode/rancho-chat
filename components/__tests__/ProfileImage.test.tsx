import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../store/store";

import ProfileImage from "../ProfileImage";

describe("<ProfileImage />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProfileImage size={50} uri="testUrl" userId="testUserId" />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
