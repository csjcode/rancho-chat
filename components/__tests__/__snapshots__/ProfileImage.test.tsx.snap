// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileImage /> renders correctly 1`] = `
<View
  onPress={[Function]}
>
  <Image
    source={
      Object {
        "uri": "testUrl",
      }
    }
    style={
      Object {
        "borderColor": "#7f8c8d",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;
